src/main/java/com/jakub/todoSandbox/repository/InMemoryRepository.java

```diff
@@ -0,0 +1,165 @@
+                    existingTodo.steps().add(createdStep);
+                } else {
+                    System.out.println("You have reached the maximum number of steps of 10 for todo with id: " + existingTodo.id());
+                    break;
+                }
+            }
+            System.out.println("Size of step array after adding new: " + existingTodo.steps().size());
+        } else {
+            System.out.println("Such a Todo does not exist.");
+        }
+    }
+
+    @Override
+    public void updateStep(Long todoId, int oldStepId, Step updatedStep) {
+        var existingTodo = todoList.get(todoId);
+        if (existingTodo != null) {
+            for (int i = 0; i < existingTodo.steps().size(); i++) {
+                if (existingTodo.steps().get(i).id() == oldStepId) {
+                    existingTodo.steps().set(i, createNewStepId(((long) oldStepId), updatedStep));
+                    System.out.println("Step with id:" + oldStepId + " has been updated");
+                    return;
+                }
+            }
+            System.out.println("Step with id:" + oldStepId + " does not exist");
+        } else {
+            System.out.println("Todo with id:" + todoId + " does not exist");
+        }
+    }
+
+    @Override
+    public void deleteSteps(Long todoId, List<Long> stepIds) {
+        var existingTodo = todoList.get(todoId);
+        if (existingTodo != null && stepIds.size() > 0 && !stepIds.contains(0L)) {
+            System.out.println("Size of step array before removing step: " + existingTodo.steps().size());
+            Iterator<Step> iterator = existingTodo.steps().iterator();
+            while (iterator.hasNext()) {
+                Step step = iterator.next();
+                if (stepIds.contains(step.id())) {
+                    iterator.remove();
+                    System.out.println("Step with id:" + step.id() + " was removed.");
+                }
+            }
+            System.out.println("Size of step array after removing step: " + existingTodo.steps().size());
+        } else {
+            System.out.println("No such todo with this id exists.");
+        }
+    }
+
+
+    private boolean canAddStepToTodo(Todo todo) {
+        return todo.steps().size() < MAX_NUM_STEPS;
+    }
+
+    private Step createNewStepForTodo(Todo todo, Step step) {
+        Long newStepId = (long) (todo.steps().size() + 1);
+        return new Step(newStepId, step.name(), step.description());
+    }
+
+    private Step createNewStepId(Long id, Step step) {
+        return new Step(id, step.name(), step.description());
+    }
+
+    private Todo createNewTodoId(Long id, Todo todo) {
+        return new Todo(id, todo.name(), todo.description(), todo.priority(), todo.steps());
+    }
+
+    private boolean validateNameAndDesc(String name, String description) {
+        if (!name.isEmpty() && !name.trim().isEmpty() && name.length() <= 100 && description.length() < 3000) {
```

Comment:
Strange. We are checking name and description at the same time and then say that name is invalid